### PR TITLE
Add infoblox package

### PR DESCRIPTION
**Pull request by mbacovsky**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.12
* [ ] 1.11
* [ ] 1.10

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---
Scratch builds:
RHEL7 nonscl: https://koji.katello.org/koji/taskinfo?taskID=519787
Fedora 24: https://koji.katello.org/koji/taskinfo?taskID=519789

### CODE REVIEW

File: rubygem-infoblox/rubygem-infoblox.spec [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+%if 0%{?rhel} == 6
+Requires: ruby(abi)
+%else
+Requires: ruby(release)
+%endif
+Requires: ruby
+Requires: ruby(rubygems)
+Requires: rubygem(faraday)
+Requires: rubygem(faraday_middleware)
```
Comment (domcleal) [inline rubygem-infoblox/rubygem-infoblox.spec]: What provides these deps on EL7?

File: rubygem-infoblox/rubygem-infoblox.spec [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/.rspec
```
Comment (domcleal) [inline rubygem-infoblox/rubygem-infoblox.spec]: Exclude .rspec/.ruby-version, or move them to -doc - suggest you exclude all of them with `%{gem_instdir}/.*`

File: rubygem-infoblox/rubygem-infoblox.spec (hunk 1 above)
Comment (mbacovsky) [inline rubygem-infoblox/rubygem-infoblox.spec]: Ah, I ddin't noticed they are not in nonscl repos, I've added both to nonscl targets. I'm not sure why faraday is in core repos and faraday_middleware in plugins so I followed the pattern.

**Comment by mbacovsky** [comment]
Updated, faraday and faraday_middleware were added to nonscl targets for RHEL7,  %{gem_instdir}/.* is now excluded in the spec

File: rubygem-infoblox/rubygem-infoblox.spec (hunk 1 above)
Comment (domcleal) [inline rubygem-infoblox/rubygem-infoblox.spec]: faraday is required by packages in the core repo, while faraday_middleware is only required by plugins, hence the split.

faraday has a dependency on multipart-post, which also isn't satisfied. Please check the package is installable.

**Comment by mbacovsky** [comment]
Updated, added multipart-post to plugin nonscl repo, moved nonscl faraday to plugin nonscl.
I was able to yum install infoblox on RHEL7

File: comps/comps-foreman-rhel7.xml [hunk 3]
```diff
@@ -195,6 +195,7 @@
       <packagereq type="default">rubygem-kafo_wizards</packagereq>
       <packagereq type="default">rubygem-little-plugger</packagereq>
       <packagereq type="default">rubygem-logging</packagereq>
+      <packagereq type="default">rubygem-multipart-post</packagereq>
```
Comment (domcleal) [inline comps/comps-foreman-rhel7.xml]: This should be in comps-foreman-plugins-rhel7.xml if you intend to keep it in the plugins repo, not this file.

**Comment by mbacovsky** [comment]
multiprt-post should be on its place now.
scratch builds for el7:
- rubygem-mulipart-post: https://koji.katello.org/koji/taskinfo?taskID=520181
- rubygem-infoblox: https://koji.katello.org/koji/taskinfo?taskID=520159
- rubygem-faraday_middleware: https://koji.katello.org/koji/taskinfo?taskID=520149
- rubygem-faraday: https://koji.katello.org/koji/taskinfo?taskID=520145

```
yum install \
"http://koji.katello.org/koji/getfile?taskID=520160&name=rubygem-infoblox-1.0.0-1.git.0.71c545b.el7.noarch.rpm" \
"http://koji.katello.org/koji/getfile?taskID=520148&name=rubygem-faraday-0.9.1-3.git.0.71c545b.el7.noarch.rpm" \
"http://koji.katello.org/koji/getfile?taskID=520150&name=rubygem-faraday_middleware-0.10.0-1.git.0.71c545b.el7.noarch.rpm" \
"http://koji.katello.org/koji/getfile?taskID=520182&name=rubygem-multipart-post-1.2.0-4.git.0.fed03d7.el7.noarch.rpm"
```

**Comment by domcleal** [comment]
Thanks, looks fine. I take it you're also going to add a package that depends on this?

**Comment by mbacovsky** [comment]
@domcleal yes, I have the packaging of `smart_proxy_dhcp_infoblox` and `smart_proxy_dns_infoblox` ready, just waiting for new release of those gems. Do you prefer me to send a PRs now?

**Comment by domcleal** [comment]
Just when it's ready, I'll leave this until they're here. Thanks.

**Comment by domcleal** [comment]
Thanks @mbacovsky. Merging now prior to branching for rpm/1.13 as it'll be easier for me than doing it after.

**Pull request merged**